gui/workorder-recheck.lua

```diff
@@ -0,0 +1,22 @@
+-- Makes selected `Active` work order `Checking`.
+--[====[
+
+gui/workorder-recheck
+=====================
+Sets the status to ``Checking`` (from ``Active``) of the selected work order (in the j-m or u-m screens). This makes the manager reevaluate its conditions.
+
+Example keybinding (put it in your ``dfhack*.init``-file):
+ ``keybinding add Alt-A@jobmanagement/Main gui/workorder-recheck``
+]====]
+local scr = dfhack.gui.getCurViewscreen()
+if df.viewscreen_jobmanagementst:is_instance(scr) then
+    local orders = df.global.world.manager_orders
+    local idx = scr.sel_idx
+    if idx < #orders then
+        orders[idx].status.active = false
+    else
+        dfhack.printerr("Invalid order selected")
+    end
+else
+    dfhack.printerr('Must be called on the manager screen (j-m or u-m)')
```

Comment:
I would generally prefer `qerror` because it propagates errors if this script is called from another script. And I realize I did it this way in `gui/manager-quantity` instead - for a small standalone script like this, I think it's unlikely that another script would use it instead of just setting the relevant field directly, so I think this way is fine.

Reply:
> I would generally prefer `qerror` because it propagates errors if this script is called from another script. And I realize I did it this way in `gui/manager-quantity` instead

It was just copy&pasted, so yeah. `qerror` is actually nicer in this case too: it automatically adds script name as prefix.

Reply:
> * I'm assuming you borrowed this from `gui/manager-quantity`, but one difference I see is that this script doesn't create any in-game UI elements (that script creates a dialog), so it should go in the top-level folder (i.e. drop the `gui/` prefix and move it up a folder)

Yeah, I was pondering on this, my reason was the script doesn't accept any parameters, and for it to work you need to navigate the UI to the correct state. It is a design-decision, so just tell me what do you prefer.

> * It would also be great if you could add this to the changelog - we have a changelog in this repo, which I think was added after your last PR. It would go under a "New Scripts" section under "Future", and using the first line of this script as a description would be fine.

I've made a change in the changelog (not 100% sure it was correct though because I've added a `##New Scripts` line).

On another note,
https://github.com/DFHack/scripts/blob/671dd3a59c66c53eddcaffbe5dc0d086327235ff/changelog.txt#L4
this should probably be `https://docs.dfhack.org/en/latest/docs/Documentation.html#changelog-syntax`, because there is no `Compile.html#build-changelog`.

Reply:
Yeah, that sounds right, but I'm not seeing the change in this PR - did you forget to add the file, maybe?

And yes, that link is out-of-date - thanks for catching it!

Reply:
Oh, you added it to a new `patch-2` branch. If you're making these changes on GitHub, you probably want to navigate to the right branch in your fork before editing files, which in this case would be https://github.com/TymurGubayev/scripts/tree/patch-1

I'll apply the change here - no worries

Reply:
> Yeah, I was pondering on this, my reason was the script doesn't accept any parameters, and for it to work you need to navigate the UI to the correct state. It is a design-decision, so just tell me what do you prefer.

I would prefer just `workorder-recheck` - I think this is more along the lines of things like `clear-smoke`, `geld`, `putontable`, - I can handle this.

Reply:
> Oh, you added it to a new `patch-2` branch: [44469d2](https://github.com/DFHack/scripts/commit/44469d2e7a55d627667d5031086c742c8e11cef8). If you're making these changes on GitHub, you probably want to navigate to the right branch in your fork before editing files, which in this case would be https://github.com/TymurGubayev/scripts/tree/patch-1

You are right, I'm doing (at least trying to do) it via GitHub, I haven't even realized I've missed the correct branch somehow. Sorry about that.